### ISSUE TITLE
TimeTravel: make use of syncdoc more robust

### ISSUE THREAD

**Issue by williamstein** [sagemathinc/cocalc#6540]
Somebody was using TimeTravel with a Jupyter notebook and somehow it showed the error "Uncaught Error: Table patches must be connected." with the crash popup.  This could happen if the ipynb sync doc session wasn't fully initialized yet, or maybe it was closed, or there was some issue with the network that hadn't finished resolving.

Instead of crashing, either provide a clear error message suggesting closing and opening the file or refreshing the browser.  In addition it might be possible to better track the state of everything to ensure the above error can't happen.

In particular, see the function init_syncdoc in src/packages/frontend/frame-editors/time-travel-editor/actions.ts.

**Comment by williamstein** [comment]
I think I found a way to reproduce this in dev mode:

1. Open a Jupyter notebook.
2. Click TimeTravel and ensure that the timetravel frame is the ONLY one open anywhere.
3. Refresh the browser, wait a minute or two, refresh again.
4. Get an error sometimes:

<img width="706" alt="image" src="https://user-images.githubusercontent.com/1276278/224463461-f4df888b-f2dd-4dba-982a-289c755ff7f7.png">

I can't get this error 100% reliably, but it's progress.

**Comment by williamstein** [comment]
I also got the patches error (with much lower probability):

<img width="847" alt="image" src="https://user-images.githubusercontent.com/1276278/224465195-fdf21aa1-f6c7-4d75-b8cb-e58f9461d963.png">

**Comment by williamstein** [comment]
I think the cause is in `frontend/frame-editors/time-travel-editor/actions.ts` in the worrisomely named function `hack_to_ensure_syncdoc_is_really_working`.  There's must be a difficult race condition that can be triggered when opening the file for TimeTravel view only.

Steps to reproduce:

1. Open a Jupyter notebook.
2. Click TimeTravel and ensure that the timetravel frame is the ONLY one open anywhere.
3. Refresh the browser repeatedly.
4. Every once in a while, get one of these errors.

Also, closing and rapidly opening the file can cause an error.

It seems like when not using timetravel everything is fine.

[...]

OK, I figured it out.  I deleted that hack_to_ensure_syncdoc_is_really_working and just properly implemented the stuff that it was trying to work around.  In particular, the actions for the TimeTravel frame weren't being cleaned up properly, but the syncdb was, which resulted in potential problems at random.  Instead, we just properly clean up the TimeTravel actions now.
